For some reason we have to invalidate the margin before we an get it to redraw.
http://bugs.activestate.com/show_bug.cgi?id=99808
Index: /Users/toddw/as/komodo-devel/src/scintilla/cocoa/ScintillaCocoa.mm
===================================================================
--- cocoa/ScintillaCocoa.mm	(revision 81135)
+++ cocoa/ScintillaCocoa.mm	(working copy)
@@ -1669,6 +1682,10 @@
  * Update fields to match scroll position after receiving a notification that the user has scrolled.
  */
 void ScintillaCocoa::UpdateForScroll() {
+  // KOMODO: Invalidate the margin, otherwise it will not update - bug 99808.
+  wMargin.InvalidateAll();
+  // KOMODO: end
+
   Point ptOrigin = GetVisibleOriginInMain();
   xOffset = ptOrigin.x;
   int newTop = Platform::Minimum(ptOrigin.y / vs.lineHeight, MaxScrollPos());
